server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsMetrics.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.node.stats;
+
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.action.admin.indices.stats.CommonStatsFlags;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a container that encapsulates the necessary information needed to indicate which node stats is requested.
+ */
+public class NodesStatsMetrics implements Writeable {
```

Comment:
Nit: Shall we rename this to match a bit more what it represents than which class is using it? What do you think?

Reply:
How about `NodesStatsRequestParameters`? This name more accurately reflects the function and purpose of this class, that is, it is used to store the parameters of the node stats request.

Reply:
I like it!

Reply:
I've finished it in 739ab8aec437c2a333d480a56a8919e1b8d9136b